src/core/device.c

```diff
@@ -859,9 +865,18 @@ static void device_enumerate(Manager *m) {
                 goto fail;
         }
 
+        /* At this point we don't really know if udev is running and has populated its DB
+         * because reload of PID1 can happen at anytime especially early during the boot
+         * process. So in order to figure this out, we simply assume that if the DB is
+         * fully populated, we should at least enumerate one device otherwise we consider
+         * that the device enumeration hasn't happened yet hence we shouldn't trust it */
+        m->honor_device_enumeration = false;
+
         FOREACH_DEVICE(e, dev) {
                 const char *sysfs;
 
+                m->honor_device_enumeration = true;
+
```

Comment:
hmm, i find this a tiny bit sloppy. I wonder if we can find a better approach here.

maybe simply make systemd-udev-trigger.service a magic service: we trust udev events exactly as long as it is up and running.

Reply:
Honestly, I don't remember all the details so I'm not sure to get your idea...